File: couponservice_cs/CouponService/Services/CuponService/ICuponService.cs
namespace CouponService.Services.CuponService;

using CouponService.Dtos;
using CouponService.Models;

public interface ICuponService{
    Task<Response<IEnumerable<Cupon>>> getCupons(int page);
    Task<Response<IEnumerable<Cupon>>> putCupon(long id, Cupon cupon);
    Task<Response<Cupon>> postCupons(CuponDto cuponDto);
    Task<Response<Cupon>> deleteCupons(long id);
    Task<Response<UserCupon>> postUserCupon(UserCuponDto userCuponDto);

}

File: couponservice_java/coupon_service/src/main/java/com/example/coupon_service/service/CuponService/ICuponService.java
package com.example.coupon_service.service.CuponService;

import java.util.Collection;

import com.example.coupon_service.domain.Cupon;
import com.example.coupon_service.domain.UserCupon;
import com.example.coupon_service.dto.CuponDto;
import com.example.coupon_service.dto.Response;
import com.example.coupon_service.dto.UserCuponDto;

public interface ICuponService {
    public abstract Response<Collection<Cupon>> get(int page);
    public abstract Response<Cupon> post(CuponDto cuponDto);
    public abstract Response<UserCupon> issue(UserCuponDto userCuponDto);
    public abstract Response<Cupon> delete(Long id);
}
